### PR TITLE
HELM-331: PerfDS React, various query fixes

### PR DESCRIPTION
**Pull request by synqotik**
# HELM-331: Performance Datasource Fixes

- fix variable interpolation for node and resource in Attribute queries
- Query Editor, get initial state from existing state (passed in props)
- encapsulate and provide strong typing for API requests
- check whether queries are hidden, filter them out before sending via API
- more comprehensive check for valid Filter queries, don't send to API if invalid
- ensure query targets added or concatenated as arrays to existing query
- various other fixes

# External References

* JIRA (Issue Tracker): [http://issues.opennms.org/browse/${JIRA-ISSUE-NUMBER}](https://opennms.atlassian.net/browse/HELM-331)
* Continuous Integration: [CircleCI](https://circleci.com/gh/OpenNMS/opennms-helm)

### CODE REVIEW

**Review by aramosvizcarra — APPROVED** [review]
Looks great!
There is something I knew that I don't see it implemented yet which is the allowing wildcards (* and | ) for attributes queries I think Resources and Atribut should allow * and | and generate multiple queries. We can address that after.

**Review by scthompsonopen — APPROVED** [review]

**Comment by synqotik** [comment]
> I think Resources and Atribut should allow * and | and generate multiple queries. We can address that after.

I *think* that this works, need to double check. I think it's handled on the server side, but we do pass `*` and `|`.

File: src/datasources/perf-ds-react/PerformanceDataSource.tsx [hunk 1]
```diff
@@ -86,69 +93,71 @@ export class PerformanceDataSource extends DataSourceApi<PerformanceQuery> {
...
                     const source = buildAttributeQuerySource(target);
-
                     const interpolationVars = collectInterpolationVariables(this.templateSrv, options.scopedVars)
                     const attributes = ['nodeId', 'resourceId', 'attribute', 'datasource', 'label']
 
                     const callback = (interpolatedSource: OnmsMeasurementsQuerySource) => {
                         if (interpolatedSource.nodeId) {
                             // Calculate the effective resource id after the interpolation
-                            interpolatedSource.resourceId = this.getRemoteResourceId(interpolatedSource.nodeId, interpolatedSource.resourceId);
+                            interpolatedSource.resourceId = getRemoteResourceId(interpolatedSource.nodeId, interpolatedSource.resourceId);
```
[the remark this answers is not in the document]
Comment (synqotik) [reply src/datasources/perf-ds-react/PerformanceDataSource.tsx:102]: Totally agree! And this is after I refactored alot of this out of this method. Will refactor in a subsequent PR.

File: src/datasources/perf-ds-react/PerformanceDataSource.tsx [hunk 2]
```diff
@@ -1,10 +1,14 @@
 import { DataFrame, DataQueryResponse, DataSourceApi, DataSourceInstanceSettings } from "@grafana/data";
+import { isNil } from 'lodash'
```
[the remark this answers is not in the document]
Comment (synqotik) [reply src/datasources/perf-ds-react/PerformanceDataSource.tsx]: Done!

**Pull request merged**